app/config.py

```diff
@@ -1,11 +1,165 @@
+    elif obs_class == 'PROGCAL':
+        return ObservationClass.PROGCAL
+    elif obs_class == 'PARTNERCAL':
+        return ObservationClass.PARTNERCAL
+    elif obs_class == 'ACQ':
+        return ObservationClass.ACQ
+    elif obs_class == 'ACQCAL':
+        return ObservationClass.ACQCAL
+    elif obs_class == 'DAYCAL':
+        return ObservationClass.DAYCAL
+    else:
+        raise ConfigurationError('Observation class', obs_class)
+
+def _parse_prg_types(prg_type: str) -> ProgramTypes:
+    """Parse Program type from config.yml
+
+    Args:
+        prg_type (str): Value from 
+
+    Raises:
+        ConfigurationError: _description_
+
+    Returns:
+        _type_: _description_
+    """
+    if prg_type == 'C':
+        return ProgramTypes.C
+    elif prg_type == 'CAL':
+        return ProgramTypes.CAL
+    elif prg_type == 'DD':
+        return ProgramTypes.DD
+    elif prg_type == 'DS':
+        return ProgramTypes.DS
+    elif prg_type== 'ENG':
+        return ProgramTypes.ENG
+    elif prg_type == 'FT':
+        return ProgramTypes.FT
+    elif prg_type == 'LP':
+        return ProgramTypes.LP
+    elif prg_type == 'Q':
+        return ProgramTypes.Q
+    elif prg_type == 'SV':
+        return ProgramTypes.SV
+    else:
+        raise ConfigurationError('Program type', prg_type)   
+
+def _parse_sites(sites: Union[str, List[str]]) -> FrozenSet[Site]:
+    """Parse Sites in config.yml
+
+    Args:
+        sites (Union[str, List[str]]): Option can be a list of sites or a single one
+
+    Returns:
+        FrozenSet[Site]: a frozen site that contains lucupy Site enums 
+            corresponding to each site.
+    """
+
+    def parse_site_specfic(site: str):
+        if site == 'GS':
+            return Site.GS
+        elif site == 'GN':
+            return Site.GN
+        else:
+            raise ConfigurationError('Missing site', site)
+    if sites == 'ALL_SITES':
+    # In case of ALL_SITES option, return lucupy alias for the set of all Site enums
+        return ALL_SITES 
+
+    if isinstance(config.scheduler.sites, list):
+        return frozenset(list(map(parse_site_specfic, config.scheduler.sites)))
+    else:
+        # Single site case
+        return frozenset([parse_site_specfic(sites)])
+
+
+@dataclass
+class CollectorConfig:
+    semesters: FrozenSet[Semester]
+    obs_classes: FrozenSet[ObservationClass]
+    program_types: FrozenSet[ProgramTypes]
+    sites: FrozenSet[Site]
+    time_slot_length: TimeDelta
+
 
 path = os.path.join(ROOT_DIR, 'config.yaml')
 config = OmegaConf.load(path)
+
+config_collector = CollectorConfig(frozenset(map(_parse_semesters, config.collector.semesters)), 
+                   frozenset(map(_parse_obs_class, config.collector.observation_classes)),
+                   frozenset(map(_parse_prg_types, config.collector.program_types)),
+                   _parse_sites(config.collector.sites),
+                   TimeDelta(config.collector.time_slot_length * u.min)
+                   )
```

Comment:
the config object is global because is used in both Collector, Process Manager and well whenever would need a configurable parameter across the whole code. is either config everything for file or just one and import when needed.

Reply:
It makes sense. I mean, we could wrap it in a `Singleton` but there's no reason for adding unnecessary complexity.